contracts/erc20/contracts/src/v08/LibERC20TokenV08.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.8.0;
+
+import "@0x/contracts-utils/contracts/src/v08/errors/LibRichErrorsV08.sol";
+import "@0x/contracts-utils/contracts/src/v08/LibBytesV08.sol";
+import "./IERC20TokenV08.sol";
+
+library LibERC20TokenV08 {
+    bytes private constant DECIMALS_CALL_DATA = hex"313ce567";
+
+    /// @dev Calls `IERC20TokenV08(token).approve()`.
+    ///      Reverts if the return data is invalid or the call reverts.
+    /// @param token The address of the token contract.
+    /// @param spender The address that receives an allowance.
+    /// @param allowance The allowance to set.
+    function compatApprove(IERC20TokenV08 token, address spender, uint256 allowance) internal {
+        bytes memory callData = abi.encodeWithSelector(token.approve.selector, spender, allowance);
```

Comment:
```suggestion
        bytes memory callData = abi.encodeCall(token.approve, (spender, allowance));
```
`abi.encodeCall` typechecks its arguments, so it's preferred over `abi.encodeWithSelector` for safety